clippy_lints/src/upper_case_acronyms.rs

```diff
@@ -0,0 +1,91 @@
+use crate::utils::span_lint_and_sugg;
+use if_chain::if_chain;
+use itertools::Itertools;
+use rustc_ast::ast::{Item, ItemKind, Variant};
+use rustc_errors::Applicability;
+use rustc_lint::{EarlyContext, EarlyLintPass, LintContext};
+use rustc_middle::lint::in_external_macro;
+use rustc_session::{declare_lint_pass, declare_tool_lint};
+use rustc_span::symbol::Ident;
+
+declare_clippy_lint! {
+    /// **What it does:** Checks for camel case name containing a capitalized acronym.
+    ///
+    /// **Why is this bad?** In CamelCase, acronyms count as one word.
+    /// See [naming conventions](https://rust-lang.github.io/api-guidelines/naming.html#casing-conforms-to-rfc-430-c-case)
+    /// for more.
+    ///
+    /// **Known problems:** When two acronyms are contiguous, they are interpreted as a one acronym.
+    ///
+    /// **Example:**
+    ///
+    /// ```rust
+    /// struct HTTPResponse;
+    /// ```
+    /// Use instead:
+    /// ```rust
+    /// struct HttpResponse;
+    /// ```
+    pub UPPER_CASE_ACRONYMS,
```

Comment:
```suggestion
    pub UPPERCASE_ACRONYMS,
```

Uppercase is usually written as one word. Like in the [`char::to_uppercase`](https://doc.rust-lang.org/std/primitive.char.html#method.to_uppercase) method

Reply:
There's `non-upper-case-globals` lint in rustc and I suppose it's reasonable to make the name of this lint similar to that.
(It seems they wanted to go consistent with `non-camel-case-types` and `non_snake_case` when naming the rustc lint)

Reply:
Well, I find that inconsistent. But lets be consistently inconsistent and keep the name as it is 👍